src/auro-input.js

```diff
@@ -106,16 +117,174 @@ export default class AuroInput extends BaseInput {
     return null
   }
 
+  connectedCallback() {
+    super.connectedCallback();
+
+    if (this.format) {
+      // eslint-disable-next-line no-unused-vars
+      const cleave = new Cleave(this, this.format);
+    }
+  }
+
+  /**
+   * @private function to return html for credit card icon
+   * @returns {string} html string
+   */
+  getCreditCardIcon() {
+    // eslint-disable-next-line no-prototype-builtins
+    return this.icon && this.format && this.format.hasOwnProperty('creditCard')
+    ? html`
+        <div class="creditCard-icon">
+         ${this.generateCreditCardIconHtml()}
+        </div>
+      ` : html``
+  }
+
+  /**
+   * @private function to determine which card icon to use
+   * @returns {string} svg html value string
+   * Based on data here: https://www.creditcardinsider.com/learn/anatomy-of-a-credit-card/#credit-card-number
+   */
+  /* eslint-disable complexity */
+  /* eslint-disable max-statements */
+  generateCreditCardIconHtml() {
+    let icon = null;
+
+    if (!this.value) {
+      icon = creditCard;
+    } else {
+      const value = this.value.trim();
```

Comment:
Can we extract the icon determination into a separate utility function? It's quite long.

Reply:
This got me thinking and I decided to change this away from the overly complicated switch statement and move it to a simple regex checker.

Much cleaner! Thanks for instigating @geoffrich.